src/util/system.cpp

```diff
@@ -933,15 +822,14 @@ bool ArgsManager::ReadConfigFiles(std::string& error, bool ignore_invalid_keys)
             }
 
             // Warn about recursive -includeconf
-            includeconf = GetArgs("-includeconf");
+            includeconf.clear();
+            add_includes(chain_id, /* skip= */ chain_includes);
+            add_includes({}, /* skip= */ default_includes);
             {
```

Comment:
(unchanged but touched by this PR): I don't understand why this needs to be its own scope. Perhaps it can be flattened in a follow-up PR?